### PR TITLE
BWS-127: add edit sc functionality (frontend)

### PR DESCRIPTION
**Pull request by tbezrukava**
Снимок экрана 2022-03-10 в 09 28 17
Снимок экрана 2022-03-10 в 09 28 59

### CODE REVIEW

File: frontend/src/services/sc-api/sc-api.service.ts [hunk 1]
```diff
@@ -63,6 +66,26 @@ class SCApi {
     );
   }
 
+  public async updateInstance(
+    payload: SCInstanceUpdateParamsDto & SCInstanceUpdateRequestDto,
+  ): Promise<SCInstanceUpdateResponseDto> {
```
Comment (Yaroslaww-1) [inline frontend/src/services/sc-api/sc-api.service.ts]: I'd rather split this param into two params:
```suggestion
  public async updateInstance(
    params: SCInstanceUpdateParamsDto,
    payload: SCInstanceUpdateRequestDto,
  ): Promise<SCInstanceUpdateResponseDto> {
```
It improves code readability imho

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

**Comment by tbezrukava** [comment]
Снимок экрана 2022-03-10 в 12 00 55

File: frontend/src/assets/img/edit-icon.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (what1s1ove) [inline frontend/src/assets/img/edit-icon.svg:1]:
```suggestion
<svg width="20" height="20" viewBox="0 0 20 20" fill="//add here our orange color" xmlns="http://www.w3.org/2000/svg">
```

File: frontend/src/components/app/app.tsx [hunk 3]
```diff
@@ -104,8 +104,23 @@ const App: FC = () => {
...
-        />
+          element={
+            <AuthorizedRoute
+              component={<SCConfigurateInstance />}
+              permissions={[Permission.MANAGE_SC]}
+            />
+          }
+        >
+          <Route
+            path=":id"
```
Comment (what1s1ove) [inline frontend/src/components/app/app.tsx]:
```suggestion
            path="AppRoute.$ID"
```

File: frontend/src/components/common/button/button.tsx [hunk 4]
```diff
@@ -10,10 +10,11 @@ import { getValidClasses } from 'helpers/helpers';
 import styles from './styles.module.scss';
 import { Link } from '../link/link';
 import deleteIcon from 'assets/img/delete-icon.svg';
+import editIcon from 'assets/img/edit-icon.svg';
 
 type Props = {
   label: string;
-  to?: AppRoute;
+  to?: AppRoute | string;
```
Comment (what1s1ove) [inline frontend/src/components/common/button/button.tsx:17]:
```suggestion
  to?: AppRoute;
```
cast a type where you need it (`anyString as AppRoute`)

File: frontend/src/components/common/icon-button/icon-button.tsx [hunk 5]
```diff
@@ -1,21 +1,23 @@
...
 import { Button } from 'components/common/common';
-import { IconName } from 'common/enums/ui/icon-name.enum';
+import { IconName, AppRoute } from 'common/enums/enums';
 
 type Props = {
   icon: IconName;
   label: string;
-  onClick: () => void;
+  onClick?: () => void;
+  to?: AppRoute | string;
```
Comment (what1s1ove) [inline frontend/src/components/common/icon-button/icon-button.tsx]:
```suggestion
  to?: AppRoute;
```

File: frontend/src/components/sc-configurate-instance/components/update-instance-form/update-instance-form.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+import { getNameOf } from 'helpers/helpers';
+import {
+  useAppDispatch,
+  useAppForm,
+  useAppSelector,
+  useEffect,
+} from 'hooks/hooks';
+import { SCConfigurateInstance as scActions } from 'store/actions';
+import { scInstanceUpdate as UpdateInstanceValidationSchema } from 'validation-schemas/validation-schemas';
+import { navigation } from 'services/services';
```
Comment (what1s1ove) [inline frontend/src/components/sc-configurate-instance/components/update-instance-form/update-instance-form.tsx]:
```suggestion
```
you can't use services directly in the components. use react-dom hooks

File: frontend/src/components/sc/components/instances-table/components/action-cell/action-cell.tsx [hunk 7]
```diff
@@ -9,7 +10,14 @@ const ActionCell = (
     onInstanceDelete(id);
   };
   return (
-    <IconButton icon={IconName.TRASH} label="Delete" onClick={handleDelete} />
+    <div className={styles.wrapper}>
+      <IconButton
+        icon={IconName.GEAR}
+        label="Edit"
+        to={`${AppRoute.SC_CONFIGURATE_INSTANCE}/${id}`}
```
Comment (what1s1ove) [inline frontend/src/components/sc/components/instances-table/components/action-cell/action-cell.tsx]:
```suggestion
        to={`${AppRoute.SC_CONFIGURATE_INSTANCE}/${id}` as AppRoute}
```

**Review by Yaroslaww-1 — DISMISSED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**